server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -1,13 +1,44 @@
 package org.http4s
 
-import scalaz.{OptionT, Kleisli}
+import scalaz.{Equal, Kleisli, Monoid}
 import scalaz.concurrent.Task
-import scalaz.std.option._
 import scalaz.syntax.kleisli._
-import scalaz.syntax.monad._
-import scalaz.syntax.traverse._
+
+package server {
+
+  /**
+    * Encapsulates the notion of fallthrough orElse for a Service
+    * For any given B, if a Fallthrough[B] exists within implicit context
+    * then [[Service#orElse]] can be used.
+    */
+  trait Fallthrough[B] {
+    def isFallthrough(a: B): Boolean
+    def fallthrough[A](fst: B, snd: Service[A, B]): Service[A, B] =
+      if (isFallthrough(fst)) snd else Service.constVal(fst)
+  }
+
+}
 
 package object server {
+
+  /** Houses the principal [[Fallthrough]] typeclass instances. */
+  object Fallthrough {
+    /** Sintacticl utility for recovering the [[Fallthrough]] currently in scope. */
+    def apply[B](implicit F : Fallthrough[B]): Fallthrough[B] = F
+
+    /** A [[Fallthrough]] for any Monoid with an Equals. */
+    implicit def forMonoid[B : Monoid : Equal]: Fallthrough[B] = new Fallthrough[B] {
+      def isFallthrough(a: B): Boolean = Monoid[B].isMZero(a)
+    }
+
+    /** A [[Response]] specific [[Fallthrough]] which considers any response with a 404
+      * status code as a fallthrough. */
+    implicit def forResponse: Fallthrough[Response] = new Fallthrough[Response] {
+      def isFallthrough(r: Response): Boolean = r.status.code == 404
+    }
+
```

Comment:
That seems fine.

Are we now agreeing that 404, vs. something with an attribute, is the default implicit fallthrough Response?